### PR TITLE
add social share wrapper

### PR DESCRIPTION
**Pull request by learykara**
Adds a component that you can use to automagically create a social share button for twitter, linkedin or facebook.  It's very un-opinionated on what the element should look like because we may want buttons vs icons at various spots throughout the site.  Component takes a `url` (facebook and linkedin) or `content` (twitter) to share 

usage:
```
<SocialShare type="twitter" content="A tweet">
  <Icon name="twitter"/>
</SocialShare>
```

### CODE REVIEW

File: src/SocialShare/SocialShare.jsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    type: React.PropTypes.oneOf(['facebook', 'twitter', 'linkedin']),
+    handleTrackClick: React.PropTypes.func
+  }
+
+  _openSocialShareWindow = (width, height, location) => {
+    window.open(location, '',
+      `menubar=no,toolbar=no,resizable=yes,scrollbars=yes,` +
+      `height=${height},width=${width},left=500`);
+    return false;
+  }
```
Comment (diezguerra) [inline src/SocialShare/SocialShare.jsx]: what does this do on mobile?

File: src/SocialShare/SocialShare.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  _coerceTwitterConfig = (data) => {
+    return {
+      url: `http://twitter.com/home/?status=${this.props.content}`,
```
Comment (diezguerra) [inline src/SocialShare/SocialShare.jsx]: why no TLS?

File: src/SocialShare/SocialShare.jsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    return {
+      url: `https://www.facebook.com/sharer/sharer.php?u=${this.props.url}`,
+      width: 600,
+      height: 300
+    }
+  }
+
+  _coerceLinkedInConfig = (data) => {
+    return {
+      url: `http://www.linkedin.com/shareArticle?mini=true&url=${this.props.url}`,
```
Comment (diezguerra) [inline src/SocialShare/SocialShare.jsx]: ditto

File: src/SocialShare/SocialShare.jsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    case 'twitter':
+      return this._coerceTwitterConfig(data);
+      break;
+    case 'facebook':
+      return this._coerceFacebookConfig(data);
+      break;
+    case 'linkedin':
+      return this._coerceLinkedInConfig(data);
+      break;
+    default: return {}
```
Comment (diezguerra) [inline src/SocialShare/SocialShare.jsx]: 47-56 needs one more indent level

File: src/SocialShare/SocialShare.jsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    }
+  }
```
Comment (diezguerra) [inline src/SocialShare/SocialShare.jsx]: You could store an array with the options and then just map and call:

```es6
let urlBuilders = {
    twitter: (data) => { .. },
    facebook: (data) => { .. },
    linkedin: (data) => { .. }
}

let config = (urlBuilders['twitter'] || () => ({}))(data);
```

**Comment by diezguerra** [comment]
:+1:

File: src/SocialShare/SocialShare.jsx (hunk 2 above)
Comment (learykara) [inline src/SocialShare/SocialShare.jsx]: what is a tls?

File: src/SocialShare/SocialShare.jsx (hunk 1 above)
Comment (learykara) [inline src/SocialShare/SocialShare.jsx]: hmm.. actually not sure. I adapted this from what @tati added to the workshop detail pages a while back. Tati, do you know?

File: src/SocialShare/SocialShare.jsx (hunk 5 above)
Comment (learykara) [inline src/SocialShare/SocialShare.jsx]: will do

File: src/SocialShare/SocialShare.jsx (hunk 4 above)
Comment (learykara) [inline src/SocialShare/SocialShare.jsx]: @ripleyaffect informed me that this is a way that people write switch statements nowadays. getting rid of this in favor of your suggestion below, but I'll let you two duke it out over indentation

Comment (diezguerra) [inline src/SocialShare/SocialShare.jsx]: open bracket { + new line, one more indent.

[you people need python!](https://www.youtube.com/watch?v=zhexf94LjTw)

Comment (learykara) [inline src/SocialShare/SocialShare.jsx]: don't look at me I just work here

**Pull request merged**